### PR TITLE
[ci-visibility] Add cypress' limitation with `after:run`

### PR DESCRIPTION
**Pull request by juan-fernandez**
### What does this PR do?
Add a known limitation to the cypress plugin for ci visibility.

### Motivation
Let users know.

### Preview

https://docs-staging.datadoghq.com/juan-fernandez/add-cypress-after-run-limitation/continuous_integration/tests/javascript/?tab=cypress#cypress-afterrun-event

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/continuous_integration/tests/javascript.md [hunk 1]
```diff
@@ -319,20 +321,20 @@ it('renders a hello world', () => {
...
-[105]: /tracing/trace_collection/custom_instrumentation/nodejs?tab=locally#adding-tags
-[106]: /real_user_monitoring/browser/#setup
-[107]: /continuous_integration/guides/rum_integration/
+[1]: https://docs.cypress.io/api/plugins/writing-a-plugin#Plugins-API
+[2]: https://docs.cypress.io/guides/core-concepts/writing-and-organizing-tests#Plugins-file
+[3]: https://docs.cypress.io/guides/references/configuration#cypress-json
+[4]: https://docs.cypress.io/guides/core-concepts/writing-and-organizing-tests#Support-file
+[5]: /tracing/trace_collection/custom_instrumentation/nodejs?tab=locally#adding-tags
+[6]: /real_user_monitoring/browser/#setup
+[7]: /continuous_integration/guides/rum_integration/
```
Comment (alai97) [inline content/en/continuous_integration/tests/javascript.md:337]: These links are nested in a tab, so it's important to have them as different anchors as the links at the bottom of the page!

```suggestion
[101]: https://docs.cypress.io/api/plugins/writing-a-plugin#Plugins-API
[102]: https://docs.cypress.io/guides/core-concepts/writing-and-organizing-tests#Plugins-file
[103]: https://docs.cypress.io/guides/references/configuration#cypress-json
[104]: https://docs.cypress.io/guides/core-concepts/writing-and-organizing-tests#Support-file
[105]: /tracing/trace_collection/custom_instrumentation/nodejs?tab=locally#adding-tags
[106]: /real_user_monitoring/browser/#setup
[107]: /continuous_integration/guides/rum_integration/
```

File: content/en/continuous_integration/tests/javascript.md [hunk 2]
```diff
@@ -319,20 +321,20 @@ it('renders a hello world', () => {
 })
 ```
 
-To create filters or `group by` fields for these tags, you must first create facets. For more information about adding tags, see the [Adding Tags][105] section of the Node.js custom instrumentation documentation.
+To create filters or `group by` fields for these tags, you must first create facets. For more information about adding tags, see the [Adding Tags][5] section of the Node.js custom instrumentation documentation.
 
 ### Cypress - RUM integration
 
-If the browser application being tested is instrumented using [Browser Monitoring][106], your Cypress test results and their generated RUM browser sessions and session replays are automatically linked. For more information, see the [Instrumenting your browser tests with RUM guide][107].
+If the browser application being tested is instrumented using [Browser Monitoring][6], your Cypress test results and their generated RUM browser sessions and session replays are automatically linked. For more information, see the [Instrumenting your browser tests with RUM guide][7].
```
Comment (alai97) [inline content/en/continuous_integration/tests/javascript.md:328]:
```suggestion
If the browser application being tested is instrumented using [Browser Monitoring][106], your Cypress test results and their generated RUM browser sessions and session replays are automatically linked. For more information, see the [Instrumenting your browser tests with RUM guide][107].
```

File: content/en/continuous_integration/tests/javascript.md [hunk 3]
```diff
@@ -319,20 +321,20 @@ it('renders a hello world', () => {
 })
 ```
 
-To create filters or `group by` fields for these tags, you must first create facets. For more information about adding tags, see the [Adding Tags][105] section of the Node.js custom instrumentation documentation.
+To create filters or `group by` fields for these tags, you must first create facets. For more information about adding tags, see the [Adding Tags][5] section of the Node.js custom instrumentation documentation.
```
Comment (alai97) [inline content/en/continuous_integration/tests/javascript.md:324]:
```suggestion
To create filters or `group by` fields for these tags, you must first create facets. For more information about adding tags, see the [Adding Tags][105] section of the Node.js custom instrumentation documentation.
```

File: content/en/continuous_integration/tests/javascript.md [hunk 4]
```diff
@@ -278,8 +279,9 @@ These are the instructions if you're using a version older than `cypress@10`.
      require('dd-trace/ci/cypress/plugin')(on, config)
    }
    {{< /code-block >}}
+   <div class="alert alert-warning"><strong>Note</strong>: Datadog requires the <a href="#cypress-afterrun-event">'after:run'</a> cypress event to work, and Cypress does not currently allow multiple <a href="">'after:run'</a> handlers. This means that if you are currently using this event, dd-trace will not work properly.</div>
 
-2. Add the following line to the **top level** of your [`supportFile`][104]:
+2. Add the following line to the **top level** of your [`supportFile`][4]:
```
Comment (alai97) [inline content/en/continuous_integration/tests/javascript.md:284]:
```suggestion
2. Add the following line to the **top level** of your [`supportFile`][104]:
```

File: content/en/continuous_integration/tests/javascript.md [hunk 5]
```diff
@@ -259,12 +259,13 @@ module.exports = defineConfig({
...
 })
 {{< /code-block >}}
+<div class="alert alert-warning"><strong>Note</strong>: Datadog requires the <a href="#cypress-afterrun-event">'after:run'</a> cypress event to work, and Cypress does not currently allow multiple <a href="">'after:run'</a> handlers. This means that if you are currently using this event, dd-trace will not work properly.</div>
 
 ### Cypress before version 10
 
 These are the instructions if you're using a version older than `cypress@10`.
 
-1. Set [`pluginsFile`][102] to `"dd-trace/ci/cypress/plugin"`, for example, through [`cypress.json`][103]:
+1. Set [`pluginsFile`][2] to `"dd-trace/ci/cypress/plugin"`, for example, through [`cypress.json`][3]:
```
Comment (alai97) [inline content/en/continuous_integration/tests/javascript.md:268]:
```suggestion
1. Set [`pluginsFile`][102] to `"dd-trace/ci/cypress/plugin"`, for example, through [`cypress.json`][103]:
```

File: content/en/continuous_integration/tests/javascript.md [hunk 6]
```diff
@@ -278,8 +279,9 @@ These are the instructions if you're using a version older than `cypress@10`.
      require('dd-trace/ci/cypress/plugin')(on, config)
    }
    {{< /code-block >}}
+   <div class="alert alert-warning"><strong>Note</strong>: Datadog requires the <a href="#cypress-afterrun-event">'after:run'</a> cypress event to work, and Cypress does not currently allow multiple <a href="">'after:run'</a> handlers. This means that if you are currently using this event, dd-trace will not work properly.</div>
```
Comment (alai97) [inline content/en/continuous_integration/tests/javascript.md:282]:
```suggestion
   <div class="alert alert-warning"><strong>Note</strong>: Datadog requires the <a href="#cypress-afterrun-event">'after:run'</a> cypress event to work, and Cypress does not allow multiple <a href="">'after:run'</a> handlers. If you are using this event, <code>dd-trace</code> will not work properly.</div>
```

File: content/en/continuous_integration/tests/javascript.md [hunk 7]
```diff
@@ -259,12 +259,13 @@ module.exports = defineConfig({
   }
 })
 {{< /code-block >}}
+<div class="alert alert-warning"><strong>Note</strong>: Datadog requires the <a href="#cypress-afterrun-event">'after:run'</a> cypress event to work, and Cypress does not currently allow multiple <a href="">'after:run'</a> handlers. This means that if you are currently using this event, dd-trace will not work properly.</div>
```
Comment (alai97) [inline content/en/continuous_integration/tests/javascript.md:262]:
```suggestion
<div class="alert alert-warning"><strong>Note</strong>: Datadog requires the <a href="#cypress-afterrun-event">'after:run'</a> cypress event to work, and Cypress does not allow multiple <a href="">'after:run'</a> handlers. If you are using this event, <code>dd-trace</code> will not work properly.</div>
```

File: content/en/continuous_integration/tests/javascript.md [hunk 8]
```diff
@@ -218,7 +218,7 @@ To create filters or `group by` fields for these tags, you must first create fac
 
 ### Cypress version 10 or later
 
-Use the Cypress API documentation to [learn how to write plugins][101] for `cypress>=10`.
+Use the Cypress API documentation to [learn how to write plugins][1] for `cypress>=10`.
```
Comment (alai97) [inline content/en/continuous_integration/tests/javascript.md:221]:
```suggestion
Use the Cypress API documentation to [learn how to write plugins][101] for `cypress>=10`.
```

File: content/en/continuous_integration/tests/javascript.md [hunk 9]
```diff
@@ -438,25 +440,29 @@ From `dd-trace>=3.15.0` and `dd-trace>=2.28.0`, CI Visibility automatically uplo
 ## Known limitations
 
 ### ES modules
-[Mocha >=9.0.0][8] uses an ESM-first approach to load test files. That means that if [ES modules][11] are used (for example, by defining test files with the `.mjs` extension), _the instrumentation is limited_. Tests are detected, but there isn't visibility into your test. For more information about ES modules, see the [Node.js documentation][9].
+[Mocha >=9.0.0][8] uses an ESM-first approach to load test files. That means that if [ES modules][9] are used (for example, by defining test files with the `.mjs` extension), _the instrumentation is limited_. Tests are detected, but there isn't visibility into your test. For more information about ES modules, see the [Node.js documentation][10].
```
Comment (alai97) [inline content/en/continuous_integration/tests/javascript.md:443]: I'm not sure if the ES modules and the Node.js documentation are supposed to share the same link or not!

```suggestion
[Mocha >=9.0.0][8] uses an ESM-first approach to load test files. That means that if [ES modules][10] are used (for example, by defining test files with the `.mjs` extension), _the instrumentation is limited_. Tests are detected, but there isn't visibility into your test. For more information about ES modules, see the [Node.js documentation][10].
```

File: content/en/continuous_integration/tests/javascript.md [hunk 10]
```diff
@@ -438,25 +440,29 @@ From `dd-trace>=3.15.0` and `dd-trace>=2.28.0`, CI Visibility automatically uplo
...
-If you want visibility into the browser process, consider using [RUM & Session Replay][10]. When using Cypress, test results and their generated RUM browser sessions and session replays are automatically linked. For more information, see the [Instrumenting your browser tests with RUM guide][11].
+If you want visibility into the browser process, consider using [RUM & Session Replay][11]. When using Cypress, test results and their generated RUM browser sessions and session replays are automatically linked. For more information, see the [Instrumenting your browser tests with RUM guide][9].
 
 ### Cypress interactive mode
 
 Cypress interactive mode (which you can enter by running `cypress open`) is not supported by CI Visibility because some cypress events, such as [`before:run`][12], are not fired. If you want to try it anyway, pass `experimentalInteractiveRunEvents: true` to the [cypress configuration file][13].
 
+### Cypress `after:run` event
+
+Datadog requires the usage of Cypress' [`after:run` event][14]. Currently, Cypress only allows a single listener for this event, so if your custom Cypress plugin requires `after:run`, it will be incompatible with `dd-trace`.
```
Comment (alai97) [inline content/en/continuous_integration/tests/javascript.md:456]:
```suggestion
Datadog requires usage of the Cypress [`after:run` event][14]. Cypress only allows a single listener for this event, so if your custom Cypress plugin requires `after:run`, it is incompatible with `dd-trace`.
```

**Review by alai97 — CHANGES_REQUESTED** [review]
Some link + copy suggestions!

**Comment by juan-fernandez** [comment]
is there any other concern @alai97 ?

**Review by alai97 — APPROVED** [review]

File: content/en/continuous_integration/tests/javascript.md [hunk 11]
```diff
@@ -259,6 +259,7 @@ module.exports = defineConfig({
   }
 })
 {{< /code-block >}}
+<div class="alert alert-warning"><strong>Note</strong>: Datadog requires the <a href="#cypress-afterrun-event">'after:run'</a> cypress event to work, and Cypress does not allow multiple <a href="">'after:run'</a> handlers. If you are using this event, <code>dd-trace</code> will not work properly.</div>
```
Comment (alai97) [inline content/en/continuous_integration/tests/javascript.md:262]:
```suggestion
<div class="alert alert-warning"> Datadog requires the <a href="#cypress-afterrun-event">after:run</a> Cypress event to work, and Cypress does not allow multiple <a href="">'after:run'</a> handlers. If you are using this event, dd-trace will not work properly.</div>
```

File: content/en/continuous_integration/tests/javascript.md [hunk 12]
```diff
@@ -278,6 +279,7 @@ These are the instructions if you're using a version older than `cypress@10`.
      require('dd-trace/ci/cypress/plugin')(on, config)
    }
    {{< /code-block >}}
+   <div class="alert alert-warning"><strong>Note</strong>: Datadog requires the <a href="#cypress-afterrun-event">'after:run'</a> cypress event to work, and Cypress does not allow multiple <a href="">'after:run'</a> handlers. If you are using this event, <code>dd-trace</code> will not work properly.</div>
```
Comment (alai97) [inline content/en/continuous_integration/tests/javascript.md:282]:
```suggestion
   <div class="alert alert-warning"> Datadog requires the <a href="#cypress-afterrun-event">'after:run'</a> Cypress event to work, and Cypress does not allow multiple <a href="">'after:run'</a> handlers. If you are using this event, <code>dd-trace</code> will not work properly.</div>
```

**Pull request merged**